exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/http/metrics/OtlpHttpMetricExporterBuilder.java

```diff
@@ -112,6 +121,25 @@ public OtlpHttpMetricExporterBuilder setClientTls(byte[] privateKeyPem, byte[] c
     return this;
   }
 
+  /**
+   * Sets the X509KeyManager to use when TLS is enabled. This will replace any existing key manager
+   * that has been configured with this method, or with setClientTls().
+   */
+  public OtlpHttpMetricExporterBuilder setKeyManager(X509KeyManager x509KeyManager) {
+    delegate.setKeyManager(x509KeyManager);
+    return this;
+  }
+
+  /**
+   * Sets the SSLSocketFactory to use when TLS is enabled. If a preconfigured SSLSocketFactory is
+   * provided, a trust manager must also be provided via setTrustManager() or
+   * setTrustedCertificates().
+   */
+  public OtlpHttpMetricExporterBuilder setSslSocketFactory(SSLSocketFactory sslSocketFactory) {
```

Comment:
Ultimately we use all these options to call `OkHttpClient#sslSocketFactory(SSLSocketFactory, X509TrustManager)`. Can we simplify the API surface area by just exposing a single method: `setSslSocketFactory(SSLSocketFactory, X509TrustManager)`.

The other methods appear to just be convenience methods that allow a caller to explicitly set parts of the SSLSocketFactory / X509KeyManager while differing to the defaults for others. But this creates a lot of API surface area that is confusing. 

Just adding one method is essentially saying: use the high level methods (`setTrustedCertificates(byte[])`, `setClientTls(byte[], byte[])`) if you want us to configure stuff for you. If our configuration is insufficient, use the low level `setSslSocketFactory(SSLSocketFactory, X509TrustManager)` instead and you (the caller) are responsible for configuring all the details.

Reply:
One person's confusing is another person's powerful. 🙃 

I agree tho, I think you're right about the api surface being confusing with this. I think since we also talked about simplifying the gRPC managed channel handling in #5246 we should do that. Let me close this for now and rebuild it after some cleanup.

Reply:
Oh, some challenges though, if we want to preserve the existing APIs...

* In the current implementation, [setTrustedCertificates(byte[])](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/logs/src/main/java/io/opentelemetry/exporter/otlp/http/logs/OtlpHttpLogRecordExporterBuilder.java#L85) serves to set up the `X509TrustManager`.
* In the current implementation, [setClientTls(byte[], byte[])](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/logs/src/main/java/io/opentelemetry/exporter/otlp/http/logs/OtlpHttpLogRecordExporterBuilder.java#L94) serves to configure the `X509KeyManager`

The key manager is used in conjunction with the trust manager to create the SSL socket factory, so that also kinda conflicts with the desire to just have a clean way of BYO SSL socket factory.

If we are cool changing the existing API, we could delete the two `byte[]` methods and just say "You have to completely bring your own SSL socket factory or managed channel". If we want to preserve, we're probably looking at some confusion or at least some deprecation.

What do you think @jack-berg ?

Reply:
> I agree tho, I think you're right about the api surface being confusing with this. I think since we also talked about simplifying the gRPC managed channel handling in https://github.com/open-telemetry/opentelemetry-java/pull/5246 we should do that. Let me close this for now and rebuild it after some cleanup.

I did a lot of the legwork required for that simplification #5287.

Reply:
> The key manager is used in conjunction with the trust manager to create the SSL socket factory, so that also kinda conflicts with the desire to just have a clean way of BYO SSL socket factory.

You _either_ BYO SSL socket factory, or use the higher level `setTrustedCertificate(byte[])` and `setClientTls(byte[], byte[])` methods, not both. I think we think of the BYO SSL socket factory method as a "catch all" for scenarios not covered out of the box. 

> If we are cool changing the existing API

I don't think we should deprecate (and we certainly can't delete) these methods. They're implied in the [exporter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md) environment variables: `OTEL_EXPORTER_OTLP_CERTIFICATE`, `OTEL_EXPORTER_OTLP_CLIENT_KEY`, `OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE`.

Reply:
See#5362 for more details.